### PR TITLE
Update dependency istio/{client-go,api} and remove mixer reference.

### PR DESCRIPTION
**Pull request by incfly**
#25956 

Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ X] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Review by howardjohn — APPROVED** [review]

**Comment by incfly** [comment]
@istio/wg-user-experience-maintainers could you take a look? The changes removing mixer part is already mostly done by @kyessenov .

**Review by kyessenov — APPROVED** [review]

**Review by hzxuzhonghu — APPROVED** [review]

**Review by therealmitchconnors — APPROVED** [review]

**Review by liamawhite — APPROVED** [review]

**Review by elfinhe — APPROVED** [review]

**Comment by incfly** [comment]
```
=== RUN   TestAnalysisWritesStatus
2020-08-05T02:40:06.473096Z	info	tf	=== BEGIN: Test: 'pilot_analysis[TestAnalysisWritesStatus]' ===
    TestAnalysisWritesStatus: analysis_test.go:75: retry.UntilSuccessOrFail: timeout while waiting (last error: expected conditions to exist, but got {[] [map[code:IST0101 documentation_url:https://istio.io/docs/reference/config/analysis/ist0101/?ref=status-controller level:Error message:Referenced gateway not found: "missing-gw"]]})
2020-08-05T02:45:06.511624Z	info	tf	=== DONE (failed):  Test: 'pilot_analysis[TestAnalysisWritesStatus] (5m0.03852325s)' ===
2020-08-05T02:45:06.511663Z	error	tf	=== Dumping Namespace default-1-16467 State...
--- FAIL: TestAnalysisWritesStatus (300.06s)
FAIL
```

This has been failing in a row. However, in previous test, I do have successful runs. https://prow.istio.io/?author=incfly&job=integ-pilot-k8s-tests_istio

I looked the API change, does not find anything special about gateway and virtual service...

**Comment by howardjohn** [comment]
@therealmitchconnors ^

**Comment by therealmitchconnors** [comment]
This change is breaking the watch on virtual service.  From the [pilot log](https://storage.googleapis.com/istio-prow/pr-logs/pull/istio_istio/25960/integ-pilot-k8s-tests_istio/17726/artifacts/pilot-analysis-328cfceb00e2467e/_suite_context/istio-state-cluster-0-178179351/istiod-67df757bfc-xhcf4-discovery.log): 

```
2020-08-05T04:22:06.187581Z	warn	k8s.io/client-go@v0.18.6/tools/cache/reflector.go:125: watch of *v1alpha3.VirtualService ended with: an error on the server ("unable to decode an event from the watch stream: unable to decode watch event: v1alpha3.VirtualService.Status: unmarshalerDecoder: json: cannot unmarshal string into Go value of type map[string]json.RawMessage, error found in #10 byte of ...|ges\":null}}|..., bigger context ...|\",\"type\":\"Reconciled\"}],\"validationMessages\":null}}|...") has prevented the request from succeeding
```

**Comment by howardjohn** [comment]
That is failing to read the status, did the api change and its incompatible?

**Comment by therealmitchconnors** [comment]
It looks to me like it can't tolerate the validationmessages being null?  Aren't all fields optional in proto3?

**Comment by jasonwzm** [comment]
I am suspecting it is due to we don't have `omitempty` json tag in status field. https://github.com/istio/client-go/blob/master/pkg/apis/networking/v1alpha3/types.gen.go#L108

**Comment by jasonwzm** [comment]
@incfly Can you update the client-go to latest? The test is passing on my local where my client-go version is the latest.

**Comment by incfly** [comment]
@jasonwzm sure. let me git it a try.

**Comment by incfly** [comment]
@jasonwzm Updated https://github.com/istio/istio/pull/25960/commits/27fb6b76ae6e8d8c30c10048ef59cb65571f3b28 Still fails.

**Comment by jasonwzm** [comment]
ok, that is weird. Looking into it.

**Comment by kyessenov** [comment]
Thanks @incfly for slogging through this.
@jasonwzm, looking forward, what's the point of repo split between client-go and API? they are both go repositories that are tightly coupled.

**Comment by jasonwzm** [comment]
@kyessenov client-go is based on api repo. One big difference is that client-go types are kubernetes types, whereas api types are istio proto types. The split was also following how kubernetes does it.

**Comment by kyessenov** [comment]
@jasonwzm technically, proto types are used purely for schema. it's conceivable to switch to native structs (kubernetes types) since there's no strong advantage over using protobuf at runtime. The only drawback I can see is that api would take on k8s as a dependency, which might be OK.

**Comment by jasonwzm** [comment]
@kyessenov Our internal data model still heavily relies on usage of proto types. Also, I am not sure if making kubernetes types as default istio api types is something we'd like to have.

**Comment by jasonwzm** [comment]
It seems that `UnmarshalJSON` used by client go cannot unmarshal `v1.Time` in IstioCondition inside Status. I need to look into why that happens. @therealmitchconnors I think we should disable this test for this PR to unblock. Once we find a solution we will add the test back. (This probably involves implementing the Status API in the meantime.) wdyt?

**Comment by therealmitchconnors** [comment]
> I think we should disable this test for this PR to unblock.
Is the unmarshaling problem affecting only the test, or the feature as a whole?  If the feature is broken by this change, we need to roll back the change...

File: tests/integration/pilot/analysis/analysis_test.go
Comment (therealmitchconnors) [inline tests/integration/pilot/analysis/analysis_test.go:48]: Are we sure the problem is the test and not the feature?  I don't see any explicit calls to client-go here...

**Review by therealmitchconnors — CHANGES_REQUESTED** [review]

**Comment by jasonwzm** [comment]
From the failing pilot logs, the client go error still exists because the configs all have status written to them. Client-go would not be able to read those configs with the current status fields in them. So to avoid status being written, we can just skip this analysis test suite entirely. @therealmitchconnors That should be able to disable the feature in tests, right?

**Comment by jasonwzm** [comment]
Mitch and I had a quick chat. We will work on a fix to this issue ASAP.

**Comment by jasonwzm** [comment]
@incfly Should I push the changes here? Let me know if you have other preferences for the code changes.

File: tests/integration/pilot/analysis/analysis_test.go
Comment (jasonwzm) [reply tests/integration/pilot/analysis/analysis_test.go]: not skipping in the fix.

**Comment by jasonwzm** [comment]
@therealmitchconnors can you take a look at the changes and approve if looks good?

**Review by therealmitchconnors — APPROVED** [review]

**Pull request merged**
